app/controllers/application_controller.rb

```diff
@@ -43,4 +45,15 @@ def destroy_user_if_current_user
     @user = User.new
   end 
   
+  private
+  
+  def extract_locale_from_accept_language_header
+    language = request.env['HTTP_ACCEPT_LANGUAGE'].presence
+    if language
+      locale = HttpAcceptLangParser.parse(language)
+      logger.debug "* Locale set using accept_language to '#{locale}'"
+      locale.in?(User::LANGUAGES.map(&:to_sym)) ? locale : nil
```

Comment:
I noticed that was better to convert it to symbol rather to string, as the `I18n.locale` is returning a symbol (somehow both, symbol and string, seem to work).